python/pyspark/sql/pandas/conversion.py

```diff
@@ -596,7 +596,7 @@ def _create_from_pandas_with_arrow(
             ]
 
         # Slice the DataFrame to be batched
-        step = -(-len(pdf) // self.sparkContext.defaultParallelism)  # round int up
+        step = self._jconf.arrowMaxRecordsPerBatch()
```

Comment:
Yeah, that's true... perf diff seems trivial in any event and seems it works around the `spark.rpc.message.maxSize` issue.

Reply:
I believe it was like this to create the same number of partitions as when arrow is disabled, although that might have changed since. If the DataFrame is split with `arrowMaxRecordsPerBatch` and a user wanted to create a certain number of partitions, then would they have to look at the size of the input and then adjust `arrowMaxRecordsPerBatch` accordingly?

Reply:
Yeah, that's true .. but I wonder if the default number of partitions is something we should consider given that it wasn't already configurable before, and `SparkSession.createDataFrame` does not expose the number of partitions too.

If they really need, users might want to create an RDD with an explicit parallelism .. we don't support this now though (see also https://github.com/apache/spark/pull/29719).

Reply:
BTW, just to extra clarify, when the pandas DataFrame is small (lower than the threshold), the number of partitions remains same (configured by `spark.sql.leafNodeDefaultParallelism` that falls back to `sparkContext.defaultParallelism` if not set).

The number of partitions is only different when the input DataFrame is large.